model.py

```diff
@@ -219,19 +200,18 @@ def _load(path):
         if optim_state is None:
             return
 
-        if self._executor is not None:
-            self._load_optimizer(optim_state)
-        else:
-            self._lazy_load_optimizer = optim_state
+        assert self._executor
```

Comment:
add assert message

Reply:
As pointed out above, clear error message should be provided when`load` is called before `prepare`.
Also "removing lazy loading" could be tricky if programs are not constructed ahead of time.
Loading optimizer without a train program would not work, which would happen if user set mode something other than `train` before calling `prepare`

Reply:
@willthefrog @heavengate  Make all programs for train、eval、test in prepare